bundles/org.eclipse.sirius.emfjson/src/main/java/org/eclipse/sirius/emfjson/utils/GsonEObjectDeserializer.java

```diff
@@ -253,7 +253,7 @@ public List<EObject> deserialize(JsonElement jsonElement, Type type, JsonDeseria
 
                     EPackage ePackage = this.getEPackage(uri);
                     if (ePackage != null) {
-                        this.resourceSet.getPackageRegistry().put(ePackage.getNsURI(), ePackage);
+                        this.packageRegistry.put(ePackage.getNsURI(), ePackage);
```

Comment:
When the resourceSet is null, `this.packageRegistry` refers to the global package registry. In that case, I don't know the consequence adding an EPackage to the global registry